lib/philomena_web/controllers/api/json/image_controller.ex

```diff
@@ -27,4 +35,72 @@ defmodule PhilomenaWeb.Api.Json.ImageController do
         render(conn, "show.json", image: image, interactions: interactions)
     end
   end
+
+  def create(conn, %{"image" => image_params} = params) do
+    user = conn.assigns.current_user
+
+    params =
+      params
+      |> Map.put("scraper_url", Map.get(params, "url"))
+      |> Map.put("anonymous", Map.get(params, "anonymous", false))
+      |> Map.put("source_url", Map.get(params, "source_url", ""))
+      |> Map.put("description", Map.get(params, "description", ""))
+      |> Map.merge(image_params)
+
+    case user do
+      nil ->
+        conn
+        |> put_status(:unauthorized)
+        |> text("")
+
+      _ ->
+        attributes = %{
+          user: user,
+          ip: decode_ip(conn.remote_ip),
+          fingerprint: "c415049"
+        }
+
+        case Images.create_image(attributes, params) do
+          {:ok, %{image: image}} ->
+            spawn(fn ->
+              Images.repair_image(image)
+            end)
+
+            # ImageProcessor.cast(image.id)
+            Images.reindex_image(image)
+            Tags.reindex_tags(image.added_tags)
+            UserStatistics.inc_stat(user, :uploads)
+
+            interactions = Interactions.user_interactions([image], user)
```

Comment:
Interactions is always empty on creation, so it doesn't actually need to be loaded; just pass an empty list.